src/lib/circuit_value.ts

```diff
@@ -75,7 +75,23 @@ abstract class CircuitValue {
     return new this(...props);
   }
 
-  static check: (x: any) => void;
+  static check<T extends new (...args: any) => any>(
+    this: T,
+    v: InstanceType<T>
+  ) {
+    const fields = (this as any).prototype._fields;
+    if (fields === undefined || fields === null) {
+      return;
+    }
+
+    for (let i = 0; i < fields.length; ++i) {
+      const [key, propType] = fields[i];
+      const value = (v as any)[key];
+      if (propType.check === undefined)
+        throw Error('bug: circuit value without .check()');
+      propType.check(value);
```

Comment:
Does this error get thrown if a CircuitValue class does not define a `check` function for itself?